server/lib/routes/get-verify-email.js

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const got = require('got');
+const logger = require('mozlog')('server.get-verify-email');
+const config = require('../configuration');
+const ravenClient = require('../../lib/raven').ravenMiddleware;
+const joi = require('joi');
+const validation = require('../validation');
+
+const fxaAccountUrl = config.get('fxaccount_url');
+const STRING_TYPE = validation.TYPES.STRING;
+
+const VERIFICATION_ENDPOINT = `${fxaAccountUrl}/v1/recovery_email/verify_code`;
+const VERIFICATION_TIMEOUT = 5000;
+// Sentry combines both Info and Error into one if same name
+const VERIFICATION_LEVEL_ERROR = 'VerificationValidationError';
+const VERIFICATION_LEVEL_INFO = 'VerificationValidationInfo';
+
+const REQUIRED_SCHEMA = {
+  'code': STRING_TYPE.alphanum().min(32).max(32).required(),
+  'uid': STRING_TYPE.alphanum().min(32).max(32).required()
+};
+
+const FULL_SCHEMA = {
+  'code': STRING_TYPE.alphanum().min(32).max(32).required(),
+  // resume token can be long, do not use the limited STRING_TYPE
+  'resume': joi.string().alphanum().optional(),
+  'service': STRING_TYPE.alphanum().max(100).optional(),
+  'uid': STRING_TYPE.alphanum().min(32).max(32).required(),
+  'utm_campaign': STRING_TYPE.alphanum().optional(),
+  'utm_content': STRING_TYPE.alphanum().optional(),
+  'utm_medium': STRING_TYPE.alphanum().optional(),
+  'utm_source': STRING_TYPE.alphanum().optional()
+};
+
+module.exports = function () {
+  return {
+    method: 'get',
+    path: '/verify_email',
+    process: function (req, res, next) {
+      req.url = '/';
+
+      const data = {
+        code: req.query.code,
+        uid: req.query.uid
+      };
+
+      joi.validate(data, REQUIRED_SCHEMA, (err) => {
```

Comment:
This is a bit different because here we are returning HTML on this endpoint if something goes wrong. If we add validate then we get an error here then we get something like this: 
![image](https://cloud.githubusercontent.com/assets/128755/24132481/3b07dc5a-0dcc-11e7-9be4-b43e1aa21481.png) I guess we can have a special handler for certain routes